core/src/main/clojure/xtdb/log.clj

```diff
@@ -247,8 +251,53 @@
                        :kafka :xtdb.kafka/log)
                      opts))
 
-(defmethod ig/init-key :xtdb/log [_ ^Log$Factory factory]
-  (.openLog factory))
+(defmethod ig/prep-key :xtdb/log [_ factory]
+  {:block-cat (ig/ref :xtdb/block-catalog)
+   :factory factory})
+
+(defn tx-id->offset [^long tx-id]
+  (bit-and tx-id (dec (bit-shift-left 1 48))))
+
+(defn tx-id->epoch [^long tx-id]
+  (bit-shift-right tx-id 48))
+
+(def out-of-sync-log-message
+  "Node failed to start due to an invalid transaction log state (%s) that does not correspond with the latest indexed transaction (epoch=%s and offset=%s).
```

Comment:
can this detail go in the operational docs instead, and a link here? reasoning being we can point people to that if they don't spot the log message, and also gives us more flexibility with formatting/level of detail.

see bottom of LogProcessor.kt for prior art